python/lsst/ts/externalscripts/base_take_twilight_flats.py

```diff
@@ -0,0 +1,454 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["BaseTakeTwilightFlats"]
+
+import abc
+import types
+import warnings
+
+import numpy as np
+import yaml
+from astropy import coordinates
+from astropy import units as u
+from astroquery.vizier import Vizier
+from lsst.ts import salobj
+from lsst.ts.standardscripts.base_block_script import BaseBlockScript
+
+try:
+    from lsst.summit.utils import ConsDbClient
+except ImportError:
+    warnings.warn("Cannot import required libraries. Script will not work.")
+
+
+class BaseTakeTwilightFlats(BaseBlockScript, metaclass=abc.ABCMeta):
+    """Base class for taking twilight flats."""
+
+    def __init__(self, index, descr="Base script for taking twilight flats.") -> None:
+        super().__init__(index, descr)
+
+        self.config = None
+
+        self.instrument_setup_time = 0.0
+
+        self.long_timeout = 30
+
+        self.latest_exposure_id = None
+
+        self.client = ConsDbClient("http://consdb-pq.consdb:8080/consdb")
+
+    @property
+    @abc.abstractmethod
+    def tcs(self):
+        raise NotImplementedError()
+
+    @property
+    @abc.abstractmethod
+    def camera(self):
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    async def configure_tcs(self):
+        """Abstract method to configure the TCS."""
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    async def configure_camera(self):
+        """Abstract method to configure the camera, to be implemented
+        in subclasses.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    async def get_sky_counts(self) -> float:
+        """Abstract method to get the median sky counts from the last image.
+
+        Returns
+        -------
+        float
+            Sky counts in electrons.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_instrument_name(self):
+        """Abstract method to be defined in subclasses to provide the
+        instrument name.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_instrument_configuration(self) -> dict:
+        """Abstract method to get the instrument configuration.
+
+        Returns
+        -------
+        dict
+            Dictionary with instrument configuration.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def get_instrument_filter(self) -> str:
+        """Abstract method to get the instrument filter configuration.
+
+        Returns
+        -------
+        str
+            Instrument filter configuration.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    async def setup_instrument(self, ra, dec) -> str:
```

Comment:
Can you rename this method to something that reflects what it is supposed to do now? maybe `track_radec_and_setup_instrument`?